controllers/slicegateway/slicegateway.go

```diff
@@ -1392,13 +1373,40 @@ func (r *SliceGwReconciler) ReconcileGatewayDeployments(ctx context.Context, sli
 		for _, deployment := range deployments.Items {
 			found, nodePortInUse := getClientGwRemotePortInUse(ctx, r.Client, sliceGw, deployment.Name)
 			if found {
-				_, foundInMap := gwClientToRemotePortMap.Load(deployment.Name)
-				if !foundInMap {
+				// Check if the portInUse is valid.
+				// It is valid only if the list of remoteNodePorts in the slicegw object contains the portInUse.
+				if !checkIfNodePortIsValid(sliceGw.Status.Config.SliceGatewayRemoteNodePorts, nodePortInUse) {
+					// Get a valid port number for this deployment
+					portNumToUpdate := allocateNodePortToClient(sliceGw.Status.Config.SliceGatewayRemoteNodePorts, &gwClientToRemotePortMap)
```

Comment:
The allocateNodePort func seems to return a port number not already stored in the port map, and returns 0 if the input list of ports and the ports stored in the map match. What if the port map already had an entry for the deployment? It would end up returning 0, right? So i think the alloc func needs to also take the name of the deployment as input. If it cannot find a free port in the list, and before returning a 0, it should check if the map has an entry for the deployment. If yes, return the value of the entry, else return 0.

Please also check for the return value. If 0, return an error.

Reply:
Great find! i actually was handling this case before (checking if its zero in the caller) and forgot to update it in the recent change i made . Will update the PR soon and let you know.

Reply:
@bharath-avesha i have updated the PR, please have a look.